code/modules/instruments/instruments/item.dm

```diff
@@ -1,320 +1,303 @@
-							/obj/item/instrument/trumpet,
-							/obj/item/instrument/saxophone,
-							/obj/item/instrument/trombone,
-							/obj/item/instrument/recorder,
-							/obj/item/instrument/harmonica,
-							/obj/item/instrument/piano_synth/headphones
-							)
-		for(var/V in templist)
-			var/atom/A = V
-			instruments[initial(A.name)] = A
-	return instruments
+//copy pasta of the space piano, don't hurt me -Pete
+/obj/item/instrument
+	name = "generic instrument"
+	force = 10
+	max_integrity = 100
+	resistance_flags = FLAMMABLE
+	icon = 'icons/obj/musician.dmi'
+	lefthand_file = 'icons/mob/inhands/equipment/instruments_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/equipment/instruments_righthand.dmi'
+	/// Our song datum.
+	var/datum/song/handheld/song
+	/// Our allowed list of instrument ids. This is nulled on initialize.
+	var/list/allowed_instrument_ids
+	/// How far away our song datum can be heard.
+	var/instrument_range = 15
+
+/obj/item/instrument/Initialize(mapload)
+	. = ..()
+	song = new(src, allowed_instrument_ids, instrument_range)
+	allowed_instrument_ids = null			//We don't need this clogging memory after it's used.
+
+/obj/item/instrument/Destroy()
+	QDEL_NULL(song)
+	return ..()
+
+/obj/item/instrument/proc/should_stop_playing(mob/user)
+	return !user.CanReach(src) || !user.canUseTopic(src, FALSE, TRUE, FALSE, FALSE)
+
+/obj/item/instrument/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] begins to play 'Gloomy Sunday'! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	return (BRUTELOSS)
+
+/obj/item/instrument/attack_self(mob/user)
+	if(!user.IsAdvancedToolUser())
+		to_chat(user, "<span class='warning'>You don't have the dexterity to do this!</span>")
+		return TRUE
+	interact(user)
+
+/obj/item/instrument/interact(mob/user)
+	ui_interact(user)
+
+/obj/item/instrument/ui_interact(mob/living/user)
+	if(!isliving(user) || user.stat || user.restrained())
+		return
+
+	user.set_machine(src)
+	song.ui_interact(user)
+
+/obj/item/instrument/violin
+	name = "space violin"
+	desc = "A wooden musical instrument with four strings and a bow. \"The devil went down to space, he was looking for an assistant to grief.\""
+	icon_state = "violin"
+	item_state = "violin"
+	hitsound = "swing_hit"
+	allowed_instrument_ids = "violin"
+
+/obj/item/instrument/violin/golden
+	name = "golden violin"
+	desc = "A golden musical instrument with four strings and a bow. \"The devil went down to space, he was looking for an assistant to grief.\""
+	icon_state = "golden_violin"
+	item_state = "golden_violin"
+	resistance_flags = LAVA_PROOF | FIRE_PROOF | ACID_PROOF
+
+/obj/item/instrument/piano_synth
+	name = "synthesizer"
+	desc = "An advanced electronic synthesizer that can be used as various instruments."
+	icon_state = "synth"
+	item_state = "synth"
+	allowed_instrument_ids = "piano"
+
+/obj/item/instrument/piano_synth/Initialize()
+	. = ..()
+	song.allowed_instrument_ids = get_allowed_instrument_ids()
```

Comment:
Maybe build a single global or SS list, instead of populating each synthesizer with one unique list?

Reply:
ye sure